### PR TITLE
revive: add enable-all-rules.

### LINKED ISSUE #2074: Revive settings missing enableAllRules
- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).
- [ ] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

<details><summary>Description of the problem</summary>

The linter-settings for revive is missing the config field `enableAllRules`

</details>

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint version
golangci-lint has version 1.41.1 built from a2074809 on 2021-06-19T16:01:50Z
```

</details>

<details><summary>Code example or link to a public repository</summary>

https://github.com/mgechev/revive/blob/master/lint/config.go#L29

https://github.com/golangci/golangci-lint/blob/afb042badcbea2eb75a191275cd7c5d8ac143ec7/pkg/config/linters_settings.go#L375

</details>

### PR DESCRIPTION
**Pull request by ldez**
Fixes #2074

### CODE REVIEW

**Review by bombsimon — APPROVED** [review]
LGTM

File: pkg/config/linters_settings.go [hunk 1]
```diff
@@ -376,6 +376,7 @@ type ReviveSettings struct {
 	IgnoreGeneratedHeader bool `mapstructure:"ignore-generated-header"`
 	Confidence            float64
 	Severity              string
+	EnableAllRules        bool `mapstructure:"enable-all-rules"`
```
Comment (invidian) [inline pkg/config/linters_settings.go:379]: Should that be documented in https://github.com/golangci/golangci-lint/blob/master/.golangci.example.yml?

Comment (ldez) [reply pkg/config/linters_settings.go:379]: yes I am working to update the revive section in the example file.

Comment (AlekSi) [reply pkg/config/linters_settings.go:379]: It is still not documented

Comment (ldez) [reply pkg/config/linters_settings.go:379]: https://github.com/golangci/golangci-lint/pull/2458

**Pull request merged**